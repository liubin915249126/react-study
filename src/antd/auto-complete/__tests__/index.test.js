import React from 'react';
import { mount } from 'enzyme';
import AutoComplete from '..';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

describe('AutoComplete', () => {
  mountTest(AutoComplete);
  rtlTest(AutoComplete);

  it('AutoComplete with custom Input render perfectly', () => {
    const wrapper = mount(
      <AutoComplete dataSource={['12345', '23456', '34567']}>
        <textarea />
      </AutoComplete>,
    );

    expect(wrapper.find('textarea').length).toBe(1);
    wrapper.find('textarea').simulate('change', { target: { value: '123' } });

    // should not filter data source defaultly
    expect(wrapper.find('.ant-select-item-option').length).toBe(3);
  });

  it('AutoComplete should work when dataSource is object array', () => {
    const wrapper = mount(
      <AutoComplete
        dataSource={[
          { text: 'text', value: 'value' },
          { text: 'abc', value: 'xxx' },
        ]}
      >
        <input />
      </AutoComplete>,
    );
    expect(wrapper.find('input').length).toBe(1);
    wrapper.find('input').simulate('change', { target: { value: 'a' } });

    // should not filter data source defaultly
    expect(wrapper.find('.ant-select-item-option').length).toBe(2);
  });

  it('AutoComplete throws error when contains invalid dataSource', () => {
    jest.spyOn(console, 'error').mockImplementation(() => undefined);
    expect(() => {
      mount(
        <AutoComplete dataSource={[() => {}]}>
          <textarea />
        </AutoComplete>,
      );
    }).toThrow();
    // eslint-disable-next-line no-console
    console.error.mockRestore();
  });

  it('legacy dataSource should accept react element option', () => {
    const wrapper = mount(<AutoComplete open dataSource={[<span key="key">ReactNode</span>]} />);
    expect(wrapper).toMatchRenderedSnapshot();
  });

  it('legacy AutoComplete.Option should be compatiable', () => {
    const wrapper = mount(
      <AutoComplete>
        <AutoComplete.Option value="111">111</AutoComplete.Option>
        <AutoComplete.Option value="222">222</AutoComplete.Option>
      </AutoComplete>,
    );
    expect(wrapper.find('input').length).toBe(1);
    wrapper.find('input').simulate('change', { target: { value: '1' } });
    expect(wrapper.find('.ant-select-item-option').length).toBe(2);
  });

  it('should not warning when getInputElement is null', () => {
    jest.spyOn(console, 'warn').mockImplementation(() => undefined);
    mount(<AutoComplete placeholder="input here" allowClear />);
    // eslint-disable-next-line no-console
    expect(console.warn).not.toBeCalled();
    // eslint-disable-next-line no-console
    console.warn.mockRestore();
  });
});
