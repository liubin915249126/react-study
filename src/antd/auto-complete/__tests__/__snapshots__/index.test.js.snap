// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete legacy dataSource should accept react element option 1`] = `
<div
  class="ant-select ant-select-auto-complete ant-select-single ant-select-open ant-select-show-search"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-expanded="true"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        role="combobox"
        type="search"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    />
  </div>
  <div>
    <div
      class="ant-select-dropdown"
      style="opacity:0;pointer-events:none"
    >
      <div>
        <div
          id="undefined_list"
          role="listbox"
          style="height:0;width:0;overflow:hidden"
        >
          <div
            aria-label="ReactNode"
            aria-selected="false"
            id="undefined_list_0"
            role="option"
          >
            key
          </div>
        </div>
        <div
          class="rc-virtual-list"
          style="position:relative"
        >
          <div
            class="rc-virtual-list-holder"
            style="max-height:256px;overflow-y:auto;overflow-anchor:none"
          >
            <div>
              <div
                class="rc-virtual-list-holder-inner"
                style="display:flex;flex-direction:column"
              >
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-active"
                  title="ReactNode"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                    ReactNode
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select:none;-webkit-user-select:none"
                    unselectable="on"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AutoComplete rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-select ant-select-rtl ant-select-auto-complete ant-select-single ant-select-show-search"
>
  <div
    class="ant-select-selector"
  >
    <span
      class="ant-select-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autocomplete="off"
        class="ant-select-selection-search-input"
        role="combobox"
        type="search"
        value=""
      />
    </span>
    <span
      class="ant-select-selection-placeholder"
    />
  </div>
</div>
`;
